modular_skyrat/modules/ashwalkers/code/buildings/ash_tendril.dm

```diff
@@ -0,0 +1,139 @@
+//this is for revitalizing/preserving regen cores
+/obj/structure/lavaland/ash_walker/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(istype(attacking_item, /obj/item/organ/internal/regenerative_core) && user.mind.has_antag_datum(/datum/antagonist/ashwalker))
+		var/obj/item/organ/internal/regenerative_core/regen_core = attacking_item
+		regen_core.preserved()
+		playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+		balloon_alert_to_viewers("[src] revitalizes [regen_core]!")
+		return
+	return ..()
+
+//this is for logging the destruction of the tendril
+/obj/structure/lavaland/ash_walker/Destroy()
+	var/compiled_string = "The [src] has been destroyed at [loc_name(src.loc)], nearest mobs are "
+	var/found_anyone = FALSE
+	for(var/mob/living/carbon/carbons_nearby in range(7))
+		compiled_string += "[key_name(carbons_nearby)],"
+		found_anyone = TRUE
+	if(!found_anyone)
+		compiled_string += "nobody."
+	log_game(compiled_string)
+	return ..()
+
+//this is for transforming a person into an ashwalker
+/obj/structure/lavaland/ash_walker/attack_hand(mob/living/user, list/modifiers)
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/human_user = user
+	if(istype(human_user.dna.species, /datum/species/lizard/ashwalker))
+		return
+	var/allow_transform = 0
+	for(var/mob/living/carbon/human/count_human in range(2, src))
+		if(!istype(count_human.dna.species, /datum/species/lizard/ashwalker))
+			continue
+		allow_transform++
+	if(allow_transform < 2)
+		balloon_alert_to_viewers("[src] rejects the request, not enough viewers!")
+		playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+		human_user.adjustBruteLoss(10)
+		return
+	else
+		balloon_alert_to_viewers("[src] reaches out to [human_user]...")
+		var/choice = tgui_alert(human_user, "Become an Ashwalker? You will abandon your previous life and body.", "Major Choice", list("Yes", "No"))
+		if(choice != "Yes")
+			balloon_alert_to_viewers("[src] feels rejected and punishes [human_user]!")
+			playsound(src, 'sound/magic/demon_consume.ogg', 50, TRUE)
+			human_user.adjustBruteLoss(50)
+			return
+		balloon_alert_to_viewers("[src] rejoices and transforms [human_user]!")
+		human_user.unequip_everything()
+		human_user.set_species(/datum/species/lizard/ashwalker)
+		human_user.underwear = "Nude"
+		human_user.update_body()
+		human_user.mind.add_antag_datum(/datum/antagonist/ashwalker)
+		if(SSmapping.level_trait(human_user.z, ZTRAIT_ICE_RUINS_UNDERGROUND) || SSmapping.level_trait(human_user.z, ZTRAIT_ICE_RUINS_UNDERGROUND))
+			ADD_TRAIT(human_user, TRAIT_NOBREATH, ROUNDSTART_TRAIT)
+			ADD_TRAIT(human_user, TRAIT_RESISTCOLD, ROUNDSTART_TRAIT)
+		ADD_TRAIT(human_user, TRAIT_PRIMITIVE, ROUNDSTART_TRAIT)
+		playsound(src, 'sound/magic/demon_dies.ogg', 50, TRUE)
+		meat_counter++
```

Comment:
I'm tired of doing these all manually, you can probably figure out the rest from this.
```suggestion

		if(SSmapping.level_trait(human_user.z, ZTRAIT_ICE_RUINS_UNDERGROUND) || SSmapping.level_trait(human_user.z, ZTRAIT_ICE_RUINS_UNDERGROUND))
			ADD_TRAIT(human_user, TRAIT_NOBREATH, ROUNDSTART_TRAIT)
			ADD_TRAIT(human_user, TRAIT_RESISTCOLD, ROUNDSTART_TRAIT)

		ADD_TRAIT(human_user, TRAIT_PRIMITIVE, ROUNDSTART_TRAIT)
		playsound(src, 'sound/magic/demon_dies.ogg', 50, TRUE)
		meat_counter++

```